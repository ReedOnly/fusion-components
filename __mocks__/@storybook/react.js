import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
