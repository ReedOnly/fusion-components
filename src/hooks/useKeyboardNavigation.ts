
import { useState } from "react";
import useEventListener from "./useEventListener";

const KEY_HANDLERS = {
    38: "onUp",
    40: "onDown",
    37: "onLeft",
    39: "onRight",
    13: "onEnter",
    27: "onEscape",
};

export type KeyBoardEvents = {
    onUp?: Function,
    onDown?: Function,
    onLeft?: Function,
    onRight?: Function,
    onEnter?: Function,
    onEscape?: Function,
}

export default (keyBoardEvents: KeyBoardEvents) => {
    const [ref, setRef] = useState<HTMLElement | null>(null);

    const handleKeyDown = e => {
        const keyCode = e.keyCode;
        const handler = keyBoardEvents[KEY_HANDLERS[keyCode]];
        if(handler) {
            e.preventDefault();
            handler(e);
        }
    }

    useEventListener(ref, "keydown", handleKeyDown, [
        keyBoardEvents,
        ref,
    ]);

    return {setRef, current:ref}
   
}