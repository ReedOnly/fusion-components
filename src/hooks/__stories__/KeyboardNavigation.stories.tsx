import * as React from 'react';
import { storiesOf } from '@storybook/react';
import useKeyboardNavigation from '../useKeyboardNavigation';

const KeyboardNavigationStory = () => {
    const listItems = ['Item 1', 'Item 2', 'Item 3'];

    const [currentItem, setCurrentItem] = React.useState<number | null>(null);
    const ref = useKeyboardNavigation({
        onDown: () => {
            setCurrentItem(
                currentItem && currentItem < listItems.length - 1 ? currentItem + 1 : currentItem
            );
        },
        onUp: () => setCurrentItem(currentItem && currentItem > 0 ? currentItem - 1 : currentItem),
    });

    return (
        <React.Fragment>
            <input
                ref={ref}
                placeholder={
                    currentItem !== null ? listItems[currentItem] : 'Click me to use navigation'
                }
                readOnly
                onClick={() => setCurrentItem(listItems.indexOf('0'))}
            />
            <ul>
                {listItems.map((item, index) => (
                    <li
                        style={{
                            listStyleType: 'none',
                            fontWeight: index === currentItem ? 600 : 200,
                        }}
                        key={index.toString()}
                    >
                        {item}
                    </li>
                ))}
            </ul>
        </React.Fragment>
    );
};

storiesOf('Hooks|KeyboardNavigation', module).add('Default', () => {
    return <KeyboardNavigationStory />;
});
