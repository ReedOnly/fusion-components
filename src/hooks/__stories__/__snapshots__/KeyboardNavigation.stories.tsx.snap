// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hooks|KeyboardNavigation Default 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <input
      placeholder="Item 1"
    />
    <ul>
      <li
        style={
          Object {
            "fontWeight": 600,
            "listStyleType": "none",
          }
        }
      >
        Item 1
      </li>
      <li
        style={
          Object {
            "fontWeight": 200,
            "listStyleType": "none",
          }
        }
      >
        Item 2
      </li>
      <li
        style={
          Object {
            "fontWeight": 200,
            "listStyleType": "none",
          }
        }
      >
        Item 3
      </li>
    </ul>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
